kafkakewl-metrics/src/main/scala/com/mwam/kafkakewl/metrics/services/HighOffsetMetricsExposer.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Marshall Wace <opensource@mwam.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.mwam.kafkakewl.metrics.services
+
+import com.mwam.kafkakewl.metrics.MainConfig
+import com.mwam.kafkakewl.metrics.domain.{KafkaTopicPartition, KafkaTopicPartitionInfo, KafkaTopicPartitionInfoChanges}
+import zio.kafka.consumer.Consumer
+import zio.{Duration, Fiber, Schedule, Task, UIO, URIO, ZIO, ZLayer}
+import zio.stream.{ZSink, ZStream}
+import zio.metrics.*
+import zio.metrics.Metric.Counter
+
+import scala.collection.immutable.Map
+
+object Helper {
+  def newTopicMetric(topic: String): Metric.Counter[Long] = Metric.counter("kafkakewl_vnext_topic_high_offset").tagged("topic", topic)
+  def newPartitionMetric(topic: String, partition: Int): Metric.Counter[Long] =
+    Metric.counter("kafkakewl_vnext_topic_partition_high_offset").tagged("topic", topic).tagged("partition", partition.toString)
+}
```

Comment:
it would be nice if the `kafkakewl_vnext` prefix would come from outside and we'd have that as a config or even just hardcoded in the code is fine as long as it's only one place (at some point we'll change this).

Btw, shouldn't it be `kafkakewl_vnext_metrics_topic_high_offset`?

If you make the `HighOffsetMetricsExposer` a service, it becomes easier to get this string fed into it as a dependency (it could be part of a small config class). Also, if it's a service, these helper methods can be just private methods in there.

Reply:
Now exists as a part of the TopicsConfig class